apps/system/js/notifications.js

```diff
@@ -403,44 +402,57 @@ var NotificationScreen = {
      */
     var dir = (detail.bidi === 'auto' || typeof detail.bidi === 'undefined') ?
       document.documentElement.dir : detail.bidi;
-
+    // As per the UX,in case for sms notification ,the notification title
+    // will be the count along with the 'new message' text,but currently in
+    // 'detail' object, name of the sender is coming. in other cases title text
+    // in 'detail' object is appropriate.
+    var cIndex = this.unReadNotificationIndex(manifestURL);
+    if (cIndex === -1) {
```

Comment:
This still wrong, so even if I'll approve it, this version of code will not be approved by David since it violates Mozilla coding style guide, so it won't be landed and thus there will be no 'future lock screen commits' until this one will be landed.

Here is some kind of explanation about indentations and the way of making PRs:

1. Initial comments from David were about changing **only** indentation of not changed code, especially (as it was in original commit) wrong changes which lead to shifting complete functions to the left. This is wrong.
2. Next comment about this particular line was in couple of words: if you adding new nesting level (wrap some code with ``'{}'``) - then you **shall update indentation**.
   * If nothing except indentation is changed after wrapping the code - you can leave comment in PR about this change - it will help reviewers understand your code.
   * If you changing some code in wrapped block, then it's OK to not leave comment or to leave comments inline the code explaining what have being changed (if you feel that such comments will help to better understand your changes) *NOTE: only way to notify reviewers about your exact changes in such case is to make 2 separate commits: first one - changing indentation, second - making additional changes in indented code. After review such commits shall be squashed into single one going to integration.* 
3. According to your commit message this PR contains 2 parts: 1 - fixes of review comments from mozilla-b2g#32749 and 2 - fixes/implementation of several new functionalities. For better understanding and for speeding up reviews, could you please in the future don't mix everything into one PR. PR1 - for comments & PR2 - for everything else could be perfect.